synapse/third_party_rules/access_rules.py

```diff
@@ -462,7 +465,8 @@ async def _on_membership_or_invite(
             # We have to freeze the room by puppeting an admin user, which we can
             # only do for local users
             if (
-                self._is_local_user(event.sender)
+                self.config.get("freeze_room_with_no_admin", False)
```

Comment:
`self.config` is not defined on `RoomAccessRules`. Plus, it's best practice to pull this option out of the config in the class's `__init__` method (so when that server is started) rather than doing so upon each request.

I'd define `self.freeze_room_with_no_admin = config.get(...)` in `__init__` and then check that value here instead.